config/routes.rb

```diff
@@ -22,6 +21,24 @@
       get "related_methods", to: "case_studies#related_methods", :as => "related_methods"
     end
   end
+  
+  resources :characteristics do
```

Comment:
This is redundant with line 9, plus you don't need nested collection routes for searching on characteristics. Same for characteristic_groups below, until you implement create/update/delete for them. But you don't need the nested search routes regardless.